### PR TITLE
feat: add label edit and delete commands

### PR DESCRIPTION
**Pull request by jstockdi**
## Summary

Rounds out `lin label` management by adding the two missing mutations. Previously the CLI could `list` and `create` labels (and attach/detach them on issues), but had no way to modify or remove a label entity itself.

- **`lin label edit <name|uuid>`** — updates a label's `--name`, `--color`, `--description`, or `--parent-id` via Linear's `issueLabelUpdate` mutation. Requires at least one field; only the provided fields are sent in the input.
- **`lin label delete <name|uuid>`** — removes a label via Linear's `issueLabelDelete` mutation.

Both commands accept either a label **name** (case-insensitive) or a **UUID**, resolved through a new `resolve_label_identifier` helper that mirrors the existing `resolve_team_identifier` / `resolve_user_identifier` / `resolve_issue_identifier` pattern.

## Changes

- `src/cli.rs` — `Edit` and `Delete` variants on `LabelCommand`
- `src/main.rs` — dispatch for both
- `src/commands/label.rs` — `edit` and `delete` handlers (edit guards against an empty update)
- `src/api/queries.rs` — `LABEL_UPDATE_MUTATION`, `LABEL_DELETE_MUTATION`
- `src/api/types.rs` — `LabelUpdateData`, `LabelDeleteData` (reuses existing `DeletePayload`)
- `src/api/resolve.rs` — `resolve_label_identifier`
- `CHANGELOG.md` — entries under `[Unreleased] > Added`

## Testing

- `cargo build` — clean
- `cargo test` — 92/92 pass
- `cargo clippy` — no warnings
- Verified `--help` output for the new subcommands

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by jstockdi — COMMENTED** [review]
## Code Review — `feat: add label edit and delete commands`

**Overview:** Adds `label edit` and `label delete`, closing the gap so labels can be fully managed (previously only `list`/`create`). The implementation follows the established patterns closely — new Clap variants, a dispatch arm, GraphQL mutation constants, response types, and a `resolve_label_identifier` helper mirroring the other `resolve_*_identifier` functions. Build/clippy/tests are green (92 tests). Nicely scoped.

### Correctness
- 🟡 **`resolve_label_identifier` can misclassify hyphenated label names as UUIDs** (`src/api/resolve.rs`). It delegates to `is_uuid` (`len > 20 && contains('-')`). Unlike team keys or issue identifiers, label names are free-text and frequently contain hyphens — e.g. `needs-more-information` (22 chars) would short-circuit as a UUID, skip name resolution, and be sent straight to the API as an `id`, producing a confusing "not found"-style error rather than a name lookup. Worth considering: try the name resolution first and only fall back to treating the input as a raw ID, or tighten `is_uuid` to a stricter UUID shape. Pre-existing heuristic, but labels are the most likely input to trip it.
- 🟢 **`.expect("resolved one label id")`** is safe given `resolve_label_names` returns exactly one id per input name or bails — but it's an invariant-coupling between the two functions. A `bail!`/`ok_or_else` would be more defensive than a panic if that contract ever changes.

### Conventions / polish
- 🔵 **`delete` echoes the raw input, not the resolved name** — `print_success("Deleted label: {label}")` prints the UUID back when the user passes one, whereas `create`/`edit` print the real label name from the payload. The delete payload has no `issueLabel`, so resolving the name before deleting would be needed to make this consistent. Minor/cosmetic.
- 🔵 **`edit` can set `parentId` but not clear it** — there's no way to un-nest a label, since `parent_id: None` is treated as "leave unchanged". Linear's `IssueLabelUpdateInput` accepts an explicit null to clear. Acceptable as a follow-up; noting for completeness.

### Risk / safety
- 🔵 **No confirmation on `delete`** — deleting a label detaches it from every issue and isn't easily reversible. There's no existing destructive-command convention in the repo to follow (this is the first `delete`), so this isn't a regression, but a `--yes`/confirmation prompt could be worth establishing here as the pattern.

### Tests
- 🔵 No tests added. The handlers are network-bound (consistent with `create`, which is also untested), but `resolve_label_identifier`'s branching is pure logic and the `is_uuid` edge case above would be cheap to lock down with a unit test.

**Verdict:** Solid, idiomatic, and consistent with the codebase. Nothing blocking; the `is_uuid`-vs-label-name interaction is the one item I'd genuinely consider addressing before merge.

<sub>🤖 Automated review via Claude Code `/review`</sub>

**Comment by jstockdi** [comment]
Addressed the `is_uuid` finding in 344bed9.

`resolve_label_identifier` now matches by **name first** and only falls back to treating the input as a raw UUID when no name matches (otherwise it bails with the available-labels list). This means a free-text label name like `needs-more-information-before-triage` resolves correctly instead of being misread as an ID. Added unit tests covering the `is_uuid` boundary.

Build / clippy / tests all green (95 tests).

The minor items (delete echoing raw input, no parent-clearing, delete confirmation) I left as-is — happy to fold any of them in if you'd like.

**Review by jstockdi — COMMENTED** [review]
## Follow-up review — resolution & destructive-delete safety

Revisiting the current state (incl. the name-first resolver fix). One finding here goes a bit deeper than the earlier pass — both items are about how a label is resolved for the new **destructive** `delete` path.

### Findings

- 🟠 **Ambiguous name resolution across teams** (`src/api/resolve.rs`). `resolve_label_identifier` matches against `fetch_all_labels(client, None)` — the whole workspace, unfiltered by team. Linear permits the same label name in multiple teams (and workspace-level labels alongside team labels), so `.find()` returns whichever happens to be first. For `label edit`/`label delete` that means `lin label delete bug` can silently mutate or remove the *wrong* `bug`. `resolve_label_names` has the same behavior, but it was previously only used for attaching labels (additive, low blast radius); wiring it into delete raises the stakes. Consider erroring on an ambiguous name (more than one match) and/or supporting a `--team` scope, so the destructive case is unambiguous.

- 🟠 **No confirmation on `label delete`** (`src/commands/label.rs`). Deleting a label detaches it from every issue and isn't recoverable from the CLI. This is the repo's first `delete` command, so there's no existing convention — but combined with the ambiguous resolution above, a single mistyped/duplicated name can quietly wipe the wrong label. A confirmation prompt (with a `--yes`/`--force` escape hatch for scripts) would be a reasonable pattern to establish here.

### Minor

- 🔵 `delete` reports `Deleted label: {label}` using the raw input, so it echoes a UUID back when one was passed, unlike `create`/`edit` which print the resolved name. You already have `all_labels` in hand during resolution — surfacing the matched name would make the success line consistent.
- 🔵 `edit` can set `--parent-id` but can't *clear* a parent (`None` is treated as "leave unchanged"). Linear's `IssueLabelUpdateInput` accepts an explicit null. Fine as a follow-up.

Nothing blocking; the name-collision-on-delete interaction is the one I'd most want addressed (or consciously accepted) before this ships.

File: src/api/resolve.rs [hunk 1]
```diff
@@ -244,6 +244,35 @@ pub async fn resolve_cycle_identifier(
...
+///
+/// Tries a case-insensitive name match first, since label names are free-text
+/// and can themselves look like a UUID to `is_uuid` (e.g. a long hyphenated
+/// name). Only when no name matches do we fall back to treating the input as a
+/// raw UUID, so a genuine ID still works without a wasted error.
+pub async fn resolve_label_identifier(client: &LinearClient, identifier: &str) -> Result<String> {
+    let all_labels = fetch_all_labels(client, None).await?;
+
+    let lower = identifier.to_lowercase();
+    if let Some(label) = all_labels.iter().find(|l| l.name.to_lowercase() == lower) {
```
Comment (jstockdi) [inline src/api/resolve.rs]: `fetch_all_labels(client, None)` is workspace-wide, and Linear allows the same label name across different teams, so this `.find()` returns an arbitrary first match. For the new `edit`/`delete` callers that means an ambiguous name can resolve to — and mutate/delete — the wrong label. Consider detecting multiple matches and bailing (or accepting a `--team` filter) before this feeds a destructive mutation.

File: src/commands/label.rs [hunk 2]
```diff
@@ -77,3 +77,64 @@ pub async fn create(
...
+    }
+
+    if let Some(label) = data.issue_label_update.issue_label {
+        output::print_success(&format!("Updated label: {}", label.name));
+    }
+
+    Ok(())
+}
+
+pub async fn delete(client: &LinearClient, label: &str) -> Result<()> {
```
Comment (jstockdi) [inline src/commands/label.rs]: Label deletion detaches the label from every issue and can't be undone from the CLI. With name resolution being ambiguous across teams (see resolve.rs), a single mistyped or duplicated name could remove the wrong label silently. Worth a confirmation prompt with a `--yes`/`--force` bypass for scripted use.

**Comment by jstockdi** [comment]
Addressed the ambiguous-name finding in f91dda5.

`resolve_label_identifier` now collects **all** case-insensitive name matches and bails when more than one is found — e.g.:

```
Label 'bug' is ambiguous — 2 labels share that name across teams: ENG, WEB.
Re-run with --team to choose one, or pass the label's UUID.
```

instead of silently mutating/deleting an arbitrary match. Added a `--team <name|key|uuid>` flag on both `label edit` and `label delete` to scope the lookup to one team. The label query/type now carry the owning team to populate that error.

Build / clippy / tests green (95). The two minor notes (delete echoing raw input, can't clear `parentId`) I've left for now — say the word if you'd like either folded in.

**Comment by jstockdi** [comment]
Superseded by #48, which bundles this label edit/delete work together with the new issue-relations and raw GraphQL commands into a single PR.
